Sources/GRPCNIOTransportCore/Server/CommonHTTP2ServerTransport.swift

```diff
@@ -241,19 +241,35 @@ package final class CommonHTTP2ServerTransport<
         return
       }
 
-      let rpcStream = RPCStream(
-        descriptor: descriptor,
-        inbound: RPCAsyncSequence(wrapping: inbound),
-        outbound: RPCWriter.Closable(
-          wrapping: ServerConnection.Stream.Outbound(
-            responseWriter: outbound,
-            http2Stream: stream
+      await withServerContextRPCCancellationHandle { handle in
+        stream.channel.eventLoop.execute {
+          // Sync is safe: this is on the right event loop.
+          let sync = stream.channel.pipeline.syncOperations
+
+          // Looking up the handler can fail if the channel is already closed, in which case
+          // cancel the handle directly.
+          do {
+            let handler = try sync.handler(type: GRPCServerStreamHandler.self)
+            handler.setCancellationHandle(handle)
+          } catch {
+            handle.cancel()
```

Comment:
In the `catch` case, is it correct to keep executing and run the `streamHandler`? Wouldn't the channel be in an inconsistent/unexpected state (or closed)?